### PR TITLE
RELEASE: Improve branching instructions

### PR DESCRIPTION
**Pull request by Turbo87**
Feedback very welcome. I hope this prevents branching problems in the future. If any issues with these instructions come up please let me know.

/cc @nathanhammond @kategengler @stefanpenner @rwjblue @kellyselden @trabus

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -27,34 +27,91 @@ $ npm --version
...
-git reset --hard origin/release
-
-# Prep the stable release
-git checkout release
-git merge beta
+# Fetch changes from GitHub
+git fetch origin
+```
+
+Once you're done following these instructions make sure that you pushed all your branches back to GitHub.
```
Comment (stefanpenner) [inline RELEASE.md]: should we be explicit as to which branches, `git pull --all` would push WIP local branches as well.

File: RELEASE.md [hunk 2]
```diff
@@ -27,34 +27,91 @@ $ npm --version
...
+# Merge any unmerged changes from "origin/beta" back in
+git merge origin/beta
+
+# ... do the beta release ...
+
+# Switch to "master" branch and reset it to "origin/master"
+git checkout -B master --track origin/master
+
+# Merge the new beta release into the "master" branch
+git merge vX.Y.0-beta.1
```
Comment (kellyselden) [inline RELEASE.md]: beta.Z?

**Comment by kellyselden** [comment]
without going through it manually, LGTM.

File: RELEASE.md (hunk 1 above)
Comment (nathanhammond) [inline RELEASE.md]: Clarification here would be:
```diff
+ push your `master`, `beta`, and `release`
- pushed all your
```

**Pull request merged**